### PR TITLE
NL: Add a tool to mine query logs

### PR DESCRIPTION
**Pull request by pradh**
Produces output [like this](https://docs.google.com/spreadsheets/d/1b-YG_vZ9j-YmnbtlNxAY3xDgaYcpjsGm3DJefuZd8Wo/edit?resourcekey=0-tGsIiFulgePUnbmdtVr20g#gid=1966584747).

### CODE REVIEW

File: tools/nl/analysis/log/mine.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+from google.cloud import bigtable
+from google.cloud.bigtable import row_filters
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('project', 'datcom-store', 'GCP project')
+flags.DEFINE_string('instance', 'website-data', 'BT instance')
+flags.DEFINE_string('table', 'nl-query', 'BT table')
+flags.DEFINE_integer('past_days', 5, 'Days to go back')
+flags.DEFINE_string('output_csv', '/tmp/nl_query_log.csv', 'Output CSV')
```
Comment (shifucun) [inline tools/nl/analysis/log/mine.py:32]: name the csv file with the mined date "2023.07.01-2023.07.05***.csv"

File: tools/nl/analysis/log/mine.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+
+def mine(table, start, csvw):
+  rows = table.read_rows(filter_=row_filters.TimestampRangeFilter(
+      row_filters.TimestampRange(start=start)))
+  for row in rows:
+    key = row.row_key.decode('utf-8')
+    # Format xxx_yyyyyyyy#<project>
+    project = key.split('#')[1].strip()
+    if project != 'datcom-website-autopush':
```
Comment (shifucun) [inline tools/nl/analysis/log/mine.py:131]: make project a constant or have a function to check this (so it's very clear where the log is from)

**Review by shifucun — APPROVED** [review]

File: tools/nl/analysis/log/mine.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      if not config:
+        continue
+
+    sentiment = fb['sentiment']
+
+    # Format xxx_yyyyyyyy#<project>
+    ts = int(key.split('#')[0].split('_')[1])
+    time = datetime.fromtimestamp(ts / 1000000.0).strftime('%Y-%m-%d %H:%M:%S')
+
+    output_row = {
```
Comment (jehangiramjad) [inline tools/nl/analysis/log/mine.py:98]: nit question: do we have access to the query url? if so, might be helpful to include it?

**Review by jehangiramjad — APPROVED** [review]
Thanks!

File: tools/nl/analysis/log/mine.py (hunk 1 above)
Comment (pradh) [reply tools/nl/analysis/log/mine.py]: good idea.

File: tools/nl/analysis/log/mine.py (hunk 3 above)
Comment (pradh) [reply tools/nl/analysis/log/mine.py:161]: great point.  added URL, also an absolute chart-index.

File: tools/nl/analysis/log/mine.py (hunk 2 above)
Comment (pradh) [reply tools/nl/analysis/log/mine.py]: done.

**Review by pradh — COMMENTED** [review]
Thanks for the reviews!

**Pull request merged**